agent/api/task.go

```diff
@@ -171,14 +171,21 @@ func (task *Task) initializeEmptyVolumes() {
 				continue
 			}
 			if _, ok := vol.(*EmptyHostVolume); ok {
-				if container.TransitionDependencySet.ContainerDependencies == nil {
-					container.TransitionDependencySet.ContainerDependencies = make([]ContainerDependency, 0)
-				}
-				container.TransitionDependencySet.ContainerDependencies = append(container.TransitionDependencySet.ContainerDependencies, ContainerDependency{
+				contDep := ContainerDependency{
 					ContainerName:   emptyHostVolumeName,
 					SatisfiedStatus: ContainerRunning,
-					DependentStatus: ContainerCreated,
-				})
+				}
+				if container.TransitionDependenciesMap == nil {
```

Comment:
why not do this outside of the for loop or when you construct the `Container` object? Let's move this outside of the `for` loop

Reply:
The container dependency is specified only in cases of empty Volumes, and this can be detected only inside the for loop. Hence we are constructing the map when required.